internal/widget/base.go

```diff
@@ -60,11 +100,17 @@ func (b *base) refresh(w fyne.Widget) {
 }
 
 func (b *base) resize(size fyne.Size, w fyne.Widget) {
-	if b.size == size {
+	var baseSize fyne.Size
+	b.readFields(func() {
+		baseSize = b.size
+	})
+	if baseSize == size {
 		return
 	}
 
+	b.propertyLock.Lock()
 	b.size = size
+	b.propertyLock.Unlock()
```

Comment:
Use setFieldsAndRefresh(f func(){...}, nil) for consistency

Reply:
But we are laying out not refreshing as a result of resize

Reply:
Again the nil means you're not refreshing. You could also define setFields(func()) if you don't like passing nil